demos/howtos/algebra.py

```diff
@@ -0,0 +1,66 @@
+"""
+Algebra in CUQIpy
+=================
+
+CUQIpy provides a simple algebraic framework for defining and manipulating
+variables.
+
+In this example, we will demonstrate how to define a simple algebraic structure
+and perform some basic algebraic operations.
+
+"""
+
+#%%
+# Defining Variables
+# ------------------
+# To utilize the algebraic framework in CUQIpy, we first need to define some
+# variables to apply the algebraic operations on. In this example, we will
+# define variables `x` and `y`.
+
+from cuqi.experimental.algebra import VariableNode
+
+x = VariableNode('x')
+y = VariableNode('y')
+
+#%%
+# Recording Algebraic Operations
+# --------------------
+# We can now perform some algebraic operations on the variables `x` and `y`.
+# The algebraic operations are recorded in a computational graph (abstract syntax tree).
+# The operations are recoding the correct ordering and adhering to the rules of
+# algebra.
+
+print("Basic operations:")
+print(x + 1)
+print(x + y)
+print(x * y)
+print(x / y)
+
+print("\nComplex operations:")
+print(x**2 + 2*x*y + y**2)
```

Comment:
small comment about printing, I think it will be nicer if there is no spaces between the power symbol and what comes before and after

x ^ 2 vs. x^2

Reply:
Fixed